theano/scalar/basic.py

```diff
@@ -681,38 +683,35 @@ def f2(f, names):
 complexs128 = _multi(complex128)
 
 
-# Using a class instead of a function makes it possible to deep-copy it in
-# Python 2.4.
-# Note that currently only a few functions use this mechanism, because it is
-# enough to make the test-suite pass with Python 2.4. However, it may prove
-# necessary to use this same mechanism in other places as well in the future.
-class upcast_out(object):
-    def __new__(self, *types):
-        dtype = Scalar.upcast(*types)
-        return get_scalar_type(dtype),
-
+# Using a class instead of a function makes it possible to deep-copy it.
+# Note that currently only a few functions use this mechanism, because
+# it is enough to make the test-suite pass. However, it may prove
+# necessary to use this same mechanism in other places as well in the
+# future.
+def upcast_out(*types):
+    dtype = Scalar.upcast(*types)
+    return get_scalar_type(dtype),
 
-class upgrade_to_float(object):
-    def __new__(self, *types):
-        """
-        Upgrade any int types to float32 or float64 to avoid losing precision.
-
-        """
-        conv = {int8: float32,
-                int16: float32,
-                int32: float64,
-                int64: float64,
-                uint8: float32,
-                uint16: float32,
-                uint32: float64,
-                uint64: float64}
-        return get_scalar_type(Scalar.upcast(*[conv.get(type, type)
-                               for type in types])),
 
+def upgrade_to_float(*types):
+    """
+    Upgrade any int types to float32 or float64 to avoid losing precision.
 
-class same_out(object):
-    def __new__(self, type):
-        return type,
+    """
+    conv = {int8: float32,
```

Comment:
Should we add `bool: float32` here as well? (NumPy upcasts it to float16, like int8).